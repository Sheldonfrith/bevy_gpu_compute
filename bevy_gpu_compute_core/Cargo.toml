[package]
name = "bevy_gpu_compute_core"
version.workspace = true
authors.workspace = true
rust-version.workspace =  true
description.workspace = true
edition.workspace = true
repository.workspace = true
license.workspace = true
keywords.workspace = true
categories.workspace = true


[dependencies]
bytemuck = {version = "1.21.0", features=["derive"]}
paste = "1.0.15"
[features]
f16 = []  # Enable unstable f16 data type support (16 bit floating point numbers), which can improve gpu compute performance if used

[dev-dependencies]
pretty_assertions = "1.4.1"
