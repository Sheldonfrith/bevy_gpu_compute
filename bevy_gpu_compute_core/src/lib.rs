#![feature(f16)]

#[allow(clippy::manual_non_exhaustive)]
// cannot use #[non_exhaustive] in a macro, and we want to force users even intra-crate to use the constructors for the matrix and vector types
mod rust;
pub mod wgsl;

pub use rust::*;
pub mod wgsl_helpers {
    pub use super::wgsl::user_facing_api::*;
}
