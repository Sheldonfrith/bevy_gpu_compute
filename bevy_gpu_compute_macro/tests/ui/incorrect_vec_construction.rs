use bevy_gpu_compute_core::wgsl_helpers::*;
const MY_CONST: Vec3Bool = Vec3Bool {
    x: true,
    y: false,
    z: true,
};
