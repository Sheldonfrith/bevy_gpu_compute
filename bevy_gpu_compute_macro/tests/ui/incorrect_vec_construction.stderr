error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/ui/incorrect_vec_construction.rs:6:3
  |
6 | };
  |   ^ consider adding a `main` function to `$DIR/tests/ui/incorrect_vec_construction.rs`

error[E0639]: cannot create non-exhaustive struct using struct expression
 --> tests/ui/incorrect_vec_construction.rs:2:28
  |
2 |   const MY_CONST: Vec3Bool = Vec3Bool {
  |  ____________________________^
3 | |     x: true,
4 | |     y: false,
5 | |     z: true,
6 | | };
  | |_^
